app/layout/header/title/component.tsx

```diff
@@ -14,22 +19,116 @@ export const Title: React.FC<TitleProps> = () => {
   const { data: projectData, isLoading: projectIsLoading } = useProject(pid);
   const { data: scenarioData, isLoading: scenarioIsLoading } = useScenario(sid);
 
-  const transitions = useTransition(!projectIsLoading && !scenarioIsLoading, null, {
+  const transitions = useTransition((!projectIsLoading && !scenarioIsLoading), null, {
     from: { opacity: 0, transform: 'translateY(-5px)' },
     enter: { opacity: 1, transform: 'translateY(0)' },
     leave: { opacity: 0, transform: 'translateY(-5px)' },
   });
 
+  // Project mutation and submit
+  const saveProjectMutation = useSaveProject({
+    requestOptions: {
+      method: 'PATCH',
+      url: `/${projectData?.id}`,
+    },
+  });
+
+  const handleProjectSubmit = useCallback(async (data) => {
+    saveProjectMutation.mutate(data, {
+      onSuccess: ({ data: s }) => {
+        console.info('Project name saved succesfully', s);
+      },
+      onError: () => {
+        console.error('Project name not saved');
+      },
+    });
+  }, [saveProjectMutation]);
+
+  // Scenario mutation and submit
+  const saveScenarioMutation = useSaveScenario({
+    requestOptions: {
+      method: 'PATCH',
+      url: `/${scenarioData?.id}`,
+    },
+  });
+
+  const handleScenarioSubmit = useCallback(async (data) => {
```

Comment:
Regarding the user experience of this feature, it feels weird to me that the focus is kept on the input after clicking `Enter`. I only knew that the update was successful because I had the developer's console open. I think that it would be more intuitive for users if the input lost its focus once the request has been completed and the update was effective.

Reply:
Ready